File: src/main/java/com/gildedrose/java/AgedBrie.java
package com.gildedrose.java;

public class AgedBrie extends Item {
    public AgedBrie(String name, int sellIn, int quality) {
        super(name, sellIn, quality);
    }

    @Override
    protected void updateQuality() {
        increaseQuality();
        if (sellIn < 0) {
            increaseQuality();
        }
    }
}

File: src/main/kotlin/com/gildedrose/kotlin/AgedBrie.kt
package com.gildedrose.kotlin

class AgedBrie(name: String, sellIn: Int, quality: Int) : Item(name, sellIn, quality) {
    override fun updateQuality() {
        increaseQuality()
        if (sellIn < 0) {
            increaseQuality()
        }
    }
}